students/becky_larson/lesson04/Mailroom_Part2.py

```diff
@@ -0,0 +1,198 @@
+    for donor, donations in donors.items():
+        print(f'\t{donor}')
+
+    print('\n')
+    return (donors)
+
+
+def add_donation(donors, donator):
+    """      Add donation to donor
+    """
+    prompt = "Please enter an amount to donate >"
+    donation = input(prompt)
+    donation = float(donation)
+    donors[donator] = donors[donator] + [donation]
+
+    return donation
+
+
+def create_card(donator, amount, fldr):
+    """
+    Create thank you note for passed user
+    """
+
+    donation_dict = {}
+    donation_dict['name'] = donator
+    donation_dict['donation'] = float(amount)
+
+    ty_text = 'Dear {name},\
+    \n\n\tThank you for your very kind donation of ${donation:.2f}.\
+    \n\n\tIt will be put to very good use.\n\n\t\t\tSincerely,\
+    \n\t\t\t   -The Team'.format(**donation_dict)
+
+    file_name = donation_dict['name'].replace(' ', '_') + '.txt'
+    file_path = os.path.join(fldr, file_name)
+
+    with open(file_path, 'w+') as f:
+        f.write(ty_text)
+
+    return True
+
+
+def send_ty(donors):
+    ''' Send thank you to selected donor '''
+
+    donation = []
+
+    prompt = "\n".join(("Enter Full name of donor",
+                        "  or Enter list to show current donors:",
+                        "  Please enter donor name:"
+                        " > "))
+    response = input(prompt)
+    response = response.title()
+
+    if response.lower() == 'list':
+        list_donors(donors)
+    elif response in donors:
+        donation = add_donation(donors, response)
+        create_card(response, donation, write_path)
+    else:
+        donors[response] = []
+        donation = add_donation(donors, response)
+        create_card(response, donation, write_path)
+
+    return donors
+
+
+def create_report(donors):
+    """
+    Print formatted report of donors and donations
+    Sort report by total donations
+    """
+
+    print("** You've selected to create a report.  **\n")
+    col1 = 'Donor Name'
+    col2 = 'Total Given'
+    col3 = 'Num Gifts'
+    col4 = 'Average Gift'
+    print(f'{col1:25} | {col2:13} | {col3:11} | {col4:13}')
+    print('-'*70)
+
+    donors = dict(sorted(donors.items(), key=sort_key, reverse=True))
+    for donor, donations in donors.items():
+        count = len(donations)
+        total = sum(donations)
+        avg = total/count
+        print(f'{donor:25}  ${total:13.2f}   {count:11}  ${avg:12.2f}')
+    return
+
+
+def ask_for_folder():
+    global user_folder
+
+    text = "\n".join(("\n\n\nEnter folder Name or hit enter for default",
+                        " >"))
+    prompt = input(text)
+
+    if prompt:
+        user_folder = prompt.replace(" ", "_")
+    else:
+        user_folder = default_folder
+
+    temp_path = tempfile.gettempdir()
+    folder_name = user_folder + '_' + today.strftime("%b_%d_%Y")
+
+    folder_path = os.path.join(temp_path, folder_name)
+
+    if not os.path.exists(folder_path):
+        os.makedirs(folder_path)
+
+    return folder_path
```

Comment:
Love this function, but not sure if it's working correctly.  Not finding the letters generated